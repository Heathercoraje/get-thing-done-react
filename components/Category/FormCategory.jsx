import React, { Component } from 'react';
import HandleClickInput from '../../lib/HandleClickInput';

const FormCategory = ({ add }) => {
	const onClickInput = evt => {
		evt.preventDefault();
		const inputElement = document.getElementById('input-cate');
		HandleClickInput(evt, add, inputElement);
	};
	return (
		<form className="form-cate" onSubmit={onClickInput}>
			<input
				placeholder=" New category"
				id="input-cate"
				style={{ display: 'none' }}
				className="input-gen input-cate"
			/>
			<input type="submit" value="+" className="button-add" />
		</form>
	);
};

export default FormCategory;
