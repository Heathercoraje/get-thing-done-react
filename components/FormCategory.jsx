import React, { Component } from 'react';

class FormCategory extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputOpen: true
    };
    this.onClickInput = this.onClickInput.bind(this);
  }

  onClickInput(event) {
    event.preventDefault();
    const name = event.target[0].value;
    this.props.add(name);
    event.target[0].value = '';
  }

  render() {
    return (
      <form onSubmit={this.onClickInput}>
        <input required placeholder="new category" />
        <input type="submit" value="+" className="button-add" />
      </form>
    );
  }
}

export default FormCategory;
