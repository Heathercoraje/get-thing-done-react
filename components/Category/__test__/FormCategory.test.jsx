import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormCategory from '../FormCategory';

test('FormCategory should render as expected', () => {
  const component = shallow(<FormCategory />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
})
