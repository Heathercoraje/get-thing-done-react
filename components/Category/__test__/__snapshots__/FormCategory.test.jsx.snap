// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCategory should render as expected 1`] = `
<form
  className="form-cate"
  onSubmit={[Function]}
>
  <input
    className="input-gen input-cate"
    id="input-cate"
    placeholder=" New category"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <input
    className="button-add"
    type="submit"
    value="+"
  />
</form>
`;
